qa/vagrant/src/main/java/org/elasticsearch/packaging/util/FileUtils.java

```diff
@@ -124,6 +130,41 @@ public static String slurp(Path file) {
         }
     }
 
+    public static String slurpTxtorGz(Path file) {
+        ByteArrayOutputStream fileBuffer = new ByteArrayOutputStream();
+        try (GZIPInputStream in = new GZIPInputStream(Channels.newInputStream(FileChannel.open(file)))) {
+            byte[] buffer = new byte[1024];
+            int len;
+
+            while ((len = in.read(buffer)) != -1) {
+                fileBuffer.write(buffer, 0, len);
+            }
+
+            return (new String(fileBuffer.toByteArray(), StandardCharsets.UTF_8));
+        } catch (ZipException e) {
+            if (e.toString().contains("Not in GZIP format")) {
+                return slurp(file);
+            }
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String slurpAllLogs(Path logPath, String activeLogFile, String rotatedLogFilesGlob) {
+        StringJoiner logFileJoiner = new StringJoiner("\n");
+        try {
+            logFileJoiner.add(new String(Files.readAllBytes(logPath.resolve(activeLogFile)), StandardCharsets.UTF_8));
+
+            for (Path rotatedLogFile : FileUtils.lsGlob(logPath, rotatedLogFilesGlob)) {
```

Comment:
We should not (probably in the javadoc) that the order of any rolled over logs is not guaranteed. In practice for the packaging tests I don't see how we would ever get more than one, but it's probably still worth noting.

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/44423/commits/69b1678433cf040575f554a23f02f4e87b2f68b3